### PR TITLE
docs(kfp): introduce multi-user KFP dependencies and KFP standalone as alternative

### PR DESCRIPTION
**Pull request by Bobgy**
**Which issue is resolved by this Pull Request:**
Resolves https://github.com/kubeflow/pipelines/issues/5452
Resolves https://github.com/kubeflow/pipelines/issues/5462

**Description of your changes:**

**Checklist:**
- [ ] Unit tests pass:
  **Make sure you have installed kustomize == 3.2.1**
    1. `make generate-changed-only`
    2. `make test`

### CODE REVIEW

**Comment by Bobgy** [comment]
/assign @zijianjoy @yanniszark @joeliedtke 

I'm updating KFP documentation on manifest root README, to clear user confusions.

File: README.md [hunk 1]
```diff
@@ -209,12 +209,31 @@ kustomize build common/istio-1-9-0/kubeflow-istio-resources/base | kubectl apply
 
 #### Kubeflow Pipelines
 
-Install the Kubeflow Pipelines official Kubeflow component:
+Install the Multi-User Kubeflow Pipelines official Kubeflow component:
 
 ```sh
 kustomize build apps/pipeline/upstream/env/platform-agnostic-multi-user | kubectl apply -f -
```
Comment (DavidSpek) [inline README.md:215]: Maybe it is a good idea to mention `apps/pipeline/upstream/env/platform-agnostic-multi-user-pns` for those that aren't using Docker as their container runtime.

**Comment by zijianjoy** [comment]
/lgtm
/approve

Thank you Yuan for the doc update!

**Comment by yanniszark** [comment]
Thanks @Bobgy!
/lgtm
/approve

**Pull request merged**